PythonVisualizations/BinaryTreeBase.py

```diff
@@ -0,0 +1,418 @@
+   def highlightCircle(self, node, color="red"):
+      self.canvas.itemconfig(node.drawable.display_shape, outline=color)
+
+   # unhighlight the circle associated with the node
+   def unHighlightCircle(self, node):
+      self.canvas.itemconfig(node.drawable.display_shape, outline="black")
+
+   # highlight the value associated with the node
+   def highlightValue(self, node, color="red"):
+      self.canvas.itemconfig(node.drawable.display_val, fill=color)
+
+   # unhighlight the value associated with the node
+   def unHighlightValue(self, node):
+      self.canvas.itemconfig(node.drawable.display_val, fill="black")
+
+   # draws the circle and the key value
+   def createNodeShape(self, x, y, key, tag):
+      circle = self.canvas.create_circle(x, y, self.CIRCLE_SIZE, tag = tag)
+      circle_text = self.canvas.create_text(x,y, text=key, font=self.VALUE_FONT, tag = tag)
+
+      # get rid of the none shape that was in its spot
+      items = self.canvas.find_overlapping(*(self.canvas.coords(circle)))
+      if len(items) >= 5:
+         self.canvas.delete(items[-3])
+         self.canvas.delete(items[-4])
+         self.canvas.delete(items[-5])
+         self.canvas.update()
+
+      return circle, circle_text
+
+   # creates the none shape that nodes will point to
+   # line coords should be the coords of the line that points to none
+   def createNoneShape(self, node, level, lineCoords, direction, tag):
+      y = lineCoords[3] + (self.CIRCLE_SIZE // 2)
+      xCoord = self.calculateCoordinates(node, level+1, direction)[0]
+
+      shrinkFactor = self.CIRCLE_SIZE // 3
+
+      coords1 = ( xCoord-self.CIRCLE_SIZE, y, 
+                  xCoord+self.CIRCLE_SIZE, y)
+      coords2 = (  shrinkFactor + coords1[0], coords1[1] + shrinkFactor,
+                  -shrinkFactor + coords1[2], coords1[3] + shrinkFactor)
+      coords3 = (  shrinkFactor + coords2[0], coords2[1] + shrinkFactor, 
+                  -shrinkFactor + coords2[2], coords2[3] + shrinkFactor)
+
+      self.canvas.create_line(coords1, tag = tag)
+      self.canvas.create_line(coords2, tag = tag)
+      self.canvas.create_line(coords3, tag = tag)
+
+   # ----------- SETTER METHODS --------------------
+   
+   # create a Node object with key and parent specified
+   # parent should be a Node object
+   def createNode(self, key, parent = None, direction = None):
+      # calculate the level
+      if not parent: level = 0
+      else: level = self.getLevel(parent) + 1                                                
+      
+      # calculate the coords
+      coords = self.calculateCoordinates(parent, level, direction)
+      x,y = coords
+
+      # generate a tag
+      tag = self.generateTag()
+
+      # create the shape and text
+      circle, circle_text = self.createNodeShape(x, y, key, tag)
+      
+      # create the drawable object
+      drawableObj = drawable(key, self.canvas.itemconfigure(circle, 'fill'), *(circle, circle_text))
```

Comment:
The call to `self.canvas.itemconfigure(circle, 'fill')` returns a tuple, and the actual fill color is the last element in the tuple.  To extract it use `self.canvas.itemconfigure(circle, 'fill')[-1]`.

Reply:
fixed with commit 9eff8cf